File: src/0275-H-Index-II/0275.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        l, n, r = 0, len(citations), len(citations)-1
        while l < r:
            mid = (l + r + 1) >> 1
            if citations[mid] <= n - mid:
                l = mid
            else:
                r = mid - 1
        if l < n and n - l > citations[l]:
            return n - l - 1
        return n - l

File: src/0275-H-Index-II/0275.cpp
class Solution 
{
public:
    int hIndex(vector<int>& citations) 
    {
        int l = 0, n = citations.size(), r = n -1;
        while (l < r)
        {
            int mid = (l + r + 1) >> 1;
            if (citations[mid] <= n - mid) l = mid;
            else r = mid - 1;
        }
        if (l < n and citations[l] < n - l) return n - l - 1;
        return n - l;
    }
};